### PR TITLE
Add support for custom read/write credentials path #87

### PR DESCRIPTION
**Pull request by sserrata**
**Usage**

```
from pancloud import Credentials

c = Credentials(path='foo')
c.get_credentials()
```
After running the code above the expectation is the `credentials.json()` file will be read/written from the relative `foo` directory.

```
from pancloud import Credentials

c = Credentials(path='/opt/pancloud/foo')
c.get_credentials()
```
After running the code above the expectation is the `credentials.json()` file will be read/written from the absolute `/opt/pancloud/foo` directory.

### CODE REVIEW

**Comment by fvigo** [comment]
I believe that we should add support for an environment variable to specify the credentials path as well.

Also, shall the variable contain the PATH or the PATH AND FILENAME?

The current behavior now is that filename is credentials.json and goes in .config under $HOME

Default: $HOME/.config/credentials.json

Options when path is specified (as $PANCREDSPATH):
A. $PANCREDSPATH (contains both path and filename, i.e. /tmp/.config/credentials.json)
B. $PANCREDSPATH/credentials.json (contains only path, i.e. /tmp/.config)
C. $PANCREDSPATH/.config/credentials.json (only overrides $HOME, i.e. /tmp)

Additional questions:
1. If the path doesn't exist, will it get created automatically or shall the operation fail? 
2. Does the answer to question 1. change depending on the adopted strategy? (A, B or C)
3. Do ".config" and "credentials.json" represent right names, or the defaults should be different? (i.e. .pancloud and .credentials.json )

**Comment by sserrata** [comment]
1. If the path doesn't exist, will it get created automatically or shall the operation fail?
- If the path doesn't exist it will be created automatically with `0o700` permissions.
2. Does the answer to question 1. change depending on the adopted strategy? (A, B or C)
- No, it should not. We check for the existence of the entire path + filename before automatically creating it.
3. Do ".config" and "credentials.json" represent right names, or the defaults should be different? (i.e. .pancloud and .credentials.json )
- We add an additional subdirectory under ".config" named "pancloud" to help distinguish us. That would make the full path `~/.config/pancloud/credentials.json`. The rationale was that `$HOME` or `%HOMEPATH%` could potentially be crowded with other "dot-whatever" files/folders and that placing the "pancloud" directory under ".config" would be the most organized approach.

**Comment by kevinsteves** [comment]
I feel this is a sufficient level of customization, to allow the
caller to change the default top level directory for credentials in
the constructor.  I don't see a need to allow a change of the
intermediate directory or store name.  I don't like the idea of
environment variables here, in the interest of simplicity and keeping
the "dials" and options minimal for security components.

File: .travis.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ python:
 - 2.7
 script:
 - tox
-- coverage run --source pancloud -m py.test
+- coverage run --source pancloud -m py.test --ignore=examples
```
Comment (sserrata) [inline .travis.yml]: Ignore `credentials_test.py` which is intermittently discovered and added to test suite by pytest

**Comment by sserrata** [comment]
Decided to add a `storage_params` argument to `Credentials` to support passing arbitrary `kwargs` to storage adapter. This solution ensures general compatibility across different storage adapters by offering a flexible way to pass adapter-specific parameters during instantiation of the `Credentials` class.

**Example**
```
from pancloud import Credentials

# Sets TinyDBStore credential path to /tmp/mycreds.json
# "dbfile" is a TinyDBStore-specific kwarg
c = Credentials(storage_params={'dbfile': '/tmp/mycreds.json'})
```

**Comment by sserrata** [comment]
#87

**Pull request merged**